runtime/in-memory-storage.js

```diff
@@ -12,6 +12,44 @@ import tracing from "../tracelib/trace.js";
 import scheduler from './scheduler.js';
 import util from './recipe/util.js';
 
+class InMemoryKey {
+  constructor(key) {
+    var parts = key.split("://");
+    this.protocol = parts[0];
```

Comment:
_protocol?

Reply:
hmm, do we need to store this here?

Reply:
This is just a simple data version of the string key.